### PR TITLE
Remove an unused parameter from eatItem and useItem

### PR DESCRIPTION
**Pull request by urkerab**
Maybe?

### CODE REVIEW

File: data/items.js [hunk 1]
```diff
@@ -4434,7 +4434,7 @@ exports.BattleItems = {
 		onAfterMoveSecondary: function (target, source, move) {
 			if (source && source !== target && source.hp && target.hp && move && move.category !== 'Status') {
 				if (!source.isActive || !this.canSwitch(source.side) || source.forceSwitchFlag || target.forceSwitchFlag) return;
-				if (target.useItem(null, source)) { // This order is correct - the item is used up even against a pokemon with Ingrain or that otherwise can't be forced out
+				if (target.useItem(source)) { // This order is correct - the item is used up even against a pokemon with Ingrain or that otherwise can't be forced out
```
Comment (Zarel) [inline data/items.js]: Does not match the function signature you define. Not that I agree with changing the API in the first place.

File: sim/pokemon.js [hunk 2]
```diff
@@ -1152,18 +1152,12 @@ class Pokemon {
 	}
 
 	/**
-	 * @param {string} itemName
 	 * @param {Pokemon} source
-	 * @param {Effect} sourceEffect
 	 */
-	useItem(itemName, source, sourceEffect) {
+	useItem(itemName, source) {
```
Comment (Zarel) [inline sim/pokemon.js]: Oh, you forgot to remove `itemName` here, it looks like you did intend to remove it.

Comment (urkerab) [reply sim/pokemon.js]: Yeah, I managed to remove the typescript for it, but not the parameter itself, oops...

File: sim/pokemon.js [hunk 3]
```diff
@@ -1178,7 +1172,7 @@ class Pokemon {
 				break;
 			}
 
-			this.battle.singleEvent('Use', item, this.itemData, this, source, sourceEffect);
+			this.battle.singleEvent('Use', item, this.itemData, this, source, this.battle.effect);
```
Comment (Zarel) [inline sim/pokemon.js]: I definitely don't agree with this instead of manually specifying `sourceEffect`.

Comment (Zarel) [reply sim/pokemon.js]: Drop `itemName` if you must, but leave `sourceEffect` alone.

**Comment by Zarel** [comment]
...@urkerab?

File: sim/pokemon.js (hunk 3 above)
Comment (urkerab) [reply sim/pokemon.js]: I don't understand what you mean by manually specifying `sourceEffect`. Do you want to leave it as an unused parameter?

(Sorry for the delayed response; I suck at time management, so I tend to acquire a backlog.)

Comment (Zarel) [reply sim/pokemon.js]: Yes, I want to leave it as an unused parameter.

Comment (Zarel) [reply sim/pokemon.js]: (Specifically: to keep our API surface consistent for mods.)

**Comment by urkerab** [comment]
I've just noticed that nobody ever passes any parameters to `eatItem`... worth removing `itemId` there too?

**Comment by panpawn** [comment]
Sidenote: I don't think commits should be past-tense, such as "Removed" - "Remove" sounds better imo. I might actually make a PR for this in CONTRIBUTING.md, maybe?

**Comment by Zarel** [comment]
`CONTRIBUTING` already says to use imperative mood, which is "Remove".

**Comment by Zarel** [comment]
(I'll remember to reword commit messages around half the time when merging, but it's usually not something I care enough about to tell someone else to reword.)

**Comment by urkerab** [comment]
Sorry, I normally write imperative commit comments, but I guess I might just have typoed it this time.

**Pull request merged**